ministryofjustice/laa-review-criminal-legal-aid
# [CRIMAPP-92] Introduce interface to allocate caseworker to work stream

## Description of change
Note this PR is the first of a couple of PR's that will enable supervisors to manage which work streams caseworkers have access to and caseworkers to work from a particular work stream. 

This PR introduces the 'Manage Competencies' interface to enable supervisors to view a list of caseworkers and their competencies (currently hardcoded). The next PR will handle the functionality to manage these competencies. 

Also adds a work stream feature flag which the tab is hidden behind. 

## Link to relevant ticket
[CRIMAPP-92](https://dsdmoj.atlassian.net/browse/CRIMAPP-92)

## Notes for reviewer
Does competencies feel right in describing what the workstreams represent in relation to a user? Skills, capabilities were also considered.

I introduced a couple of shared examples refactoring some duplicate spec testing around the pagination and ordering of a table (of users in this case). They could probably be made even more flexible but I'm not really sure if there will be many specs using these 

## Screenshots of changes (if applicable)

### Before changes:

### After changes:
<img width="1313" alt="Screenshot 2023-11-09 at 17 07 12" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/088789ed-90e5-4be5-ad48-acc7f4e8867d">
<img width="1313" alt="Screenshot 2023-11-09 at 17 07 35" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/2b046805-e61d-41ba-b9a4-a271c7610bcf">

## How to manually test the feature
Requires you to have at least one supervisor and caseworker user. 
- Log into review as a supervisor. You should see the manage competencies tab and be able to click onto it to view the dashboard. There you should be able to see a list any active caseworkers with a hardcoded competency list
- Log into review as a non supervisor role, you should not be able to view the manage competencies tab and if you navigate to `/manage_competencies`, you should see an error 
- You can also test feature flag functionality by changing your local `settings.yml` value to false and rebuilding your local server. Check you can't see the tab when logged in as a supervisor.


[CRIMAPP-92]: https://dsdmoj.atlassian.net/browse/CRIMAPP-92?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
